index.js

```diff
@@ -86,6 +86,36 @@ function compatibleMessage(thrown, errMatcher) {
 }
 
 /**
+ * ### .getFunctionName(constructorFn)
+ *
+ * Returns the name of a function.
+ * This also includes a polyfill function if `constructorFn.name` is not defined.
+ *
+ * @name getFunctionName
+ * @param {Function} constructorFn
+ * @namespace Utils
+ * @api private
+ */
+
+function getFunctionName(constructorFn) {
+  var name = '';
+  if (typeof constructorFn.name === 'undefined') {
+    // Here we run a polyfill if constructorFn.name is not defined
+    var fnNamePrefixRegex = /^[\S\s]*?constructorFntion\s*/;
```

Comment:
@keithamus @meeber great catches, guys.
Did this late yesterday, so these things have slipped out.
It's always good to have you eyes on this.

(And nice regexp "magic" there Keith :smile: )